projects/igniteui-angular-elements/src/app/custom-strategy.ts

```diff
@@ -66,9 +66,22 @@ class IgxCustomNgElementStrategy extends ComponentNgElementStrategy {
         (this as any).componentRef = {};
 
         const toBeOrphanedChildren = Array.from(element.children).filter(x => !this._componentFactory.ngContentSelectors.some(sel => x.matches(sel)));
+        const contentChildren = Array.from(element.children).filter(x => this._componentFactory.ngContentSelectors.some(sel => x.matches(sel)));
         for (const iterator of toBeOrphanedChildren) {
             // TODO: special registration OR config for custom
         }
+
+        const promises = [];
+        for (const contentChild of contentChildren) {
+            // these resolve after the parent, both the components needs to be initialized and the parent query schedule should complete before it is really ready.
+            const child = contentChild as IgcNgElement;
+            const promiseComponentReady = this.waitForCondition(() => (child.ngElementStrategy as any)?.componentRef && this.schedule.size == 0);
+            promises.push(promiseComponentReady);
+        }
+        Promise.all(promises).then(() => {
+            (this as any).componentRef?.instance.contentChildrenReady.emit();
```

Comment:
This will trigger for all components with content children - you can check the config but all column layouts & groups, toolbar, action strip at least. Those don't have `contentChildrenReady` and will actually throw here, besides it being an unnecessary work. I think the zone is doing some supper annoying consuming of promise rejections (-.-) but if you try-catch this and log the errors you'll get a bunch of those:
![image](https://github.com/user-attachments/assets/277a12da-43dc-49ee-9e69-2ed2d5ab3b21)

This is why I want the special event to be part of the config, so the strategy can check which components should emit.

Reply:
Oh, I sort of imagined the config having some typed section that had the content children loaded event name so that's read from there too. Possibly overcomplicating it, since the emit will be strategy code and possibly we won't care for different names, but still the event belongs to the component and would like to have as little as possible coupling.
But I'd be fine w/ just the check for this PR, sure.